<a href="https://colab.research.google.com/github/JBadawood/SAFCSP_Markdown/blob/main/Markdown101/Markup_Languages.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Markdown 101 | Star منصة سَطر
https://satr.codes/courses/mearAaYiwE/view

## **1. مقدمة في Markdown**
###الفروقات بين Markdown و Markup Languages

دليل الماركداون\
https://guide.dawin.io/

\
محرر دوّن\
https://www.markdownguide.org/tools/dawin/

# **Markdown**

# This is a level one header
This is a paragraph
## This a level two header
Here is a paragraph that is followed by
* an unnumbered
* list
1. and a numbered
2. list that is
3. three items long



```
# This is a level one header
This is a paragraph
## This a level two header
Here is a paragraph that is followed by
* an unnumbered
* list
1. and a numbered
2. list that is
3. three items long
```



#**HTML**



```
<h1>This is a level one header</h1>
<p>This is a paragraph</p>
<h2>This a level two header</h2>
<p>Here is a paragraph that is followed by</p>
<ul>
<li>an unnumbered</li>
<li>list</li>
</ul>
<ol type="1">
<li>and a numbered</li>
<li>list that is</li>
<li>three items long</li>
</ol>
```



<h1>This is a level one header</h1>

<p>This is a paragraph</p>

<h2>This a level two header</h2>

<p>Here is a paragraph that is followed by</p>

<ul>
<li>an unnumbered</li>
<li>list</li>
</ul>

<ol type="1">
<li>and a numbered</li>
<li>list that is</li>
<li>three items long</li>
</ol>

#**LaTeX**



```
\section{This is a level one header}
This is a paragraph
\subsection{This a level two header}
Here is a paragraph that is followed by
\begin{itemize}
\iteman unnumbered
\item list
\end{itemize}
\begin{enumerate}
\item and a numbered
\item list that is
\item three items long
\end{enumerate}

```





In [5]:
from IPython.display import display, Math, Latex, HTML

In [6]:
display(HTML("<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.3/"
               "latest.js?config=default'></script>"))

Math(r'F(k) = \int_{-\infty}^{\infty} f(x) e^{2\pi i k} dx')

<IPython.core.display.Math object>

> from:
https://stackoverflow.com/questions/48856683/latex-equations-do-not-render-in-google-colaboratory-when-using-ipython-display



```
# This is formatted as code

$F(k) = \int_{-\infty}^{\infty} f(x) e^{2\pi i k} dx$
```


$F(k) = \int_{-\infty}^{\infty} f(x) e^{2\pi i k} dx$

طريقة تحويل كود بايثون عادي لعمليات رياضية إلى تنسيق LaTeX مع إظهار النواتج

In [15]:
%pip install handcalcs

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.2/89.2 KB 3.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 274.8/274.8 KB 11.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 KB 14.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 72.0/72.0 KB 8.0 MB/s eta 0:00:00
  Attempting uninstall: mistune
    Found existing installation: mistune 0.8.4
    Uninstalling mistune-0.8.4:
      Successfully uninstalled mistune-0.8.4
  Attempting uninstall: jinja2
    Found existing installation: Jinja2 2.11.3
    Uninstalling Jinja2-2.11.3:
      Successfully uninstalled Jinja2-2.11.3
  Attempting uninstall: nbconvert
    Found existing installation: nbconvert 5.6.1
    Uninstalling nbconvert-5.6.1:
      Successfully uninstalled nbconvert-5.6.1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installe

In [16]:
import handcalcs.render

In [24]:
a = 23
b = 43
c = 2
d = 3.226

f = d / a + b

In [25]:
f

43.14026086956522

In [17]:
%%render
a = 23
b = 43
c = 2
d = 3.226

f = d / a + b

<IPython.core.display.Latex object>



```
\begin{aligned}
a &= 23 \;\textit{    }\\[10pt]
b &= 43 \;\textit{    }\\[10pt]
c &= 2 \;\textit{    }\\[10pt]
d &= 3.226 \;\textit{    }\\[10pt]
f &= \frac{ d }{ a } + b = \frac{ 3.226 }{ 23 } + 43 &= 43.14 \;\textit{    }
\end{aligned}
```




\begin{aligned}
a &= 23 \;\textit{    }\\[10pt]
b &= 43 \;\textit{    }\\[10pt]
c &= 2 \;\textit{    }\\[10pt]
d &= 3.226 \;\textit{    }\\[10pt]
f &= \frac{ d }{ a } + b = \frac{ 3.226 }{ 23 } + 43 &= 43.14 \;\textit{    }
\end{aligned}


>from:
https://stackoverflow.com/questions/48856683/latex-equations-do-not-render-in-google-colaboratory-when-using-ipython-display